### ISSUE TITLE
Odrive pro

### ISSUE THREAD

**Issue by shipeng-liu** [Factor-Robotics/odrive_ros2_control#24]
is there any plan for odrive pro?

**Comment by KongCarl** [comment]
I assume by this request Odrive Pro is not supported? Odrive is now discontinued.

**Comment by ryanpennings** [comment]
The pro isn't supported yet. I was hoping to add support early this year for a project but can't guarantee anything sorry.

**Comment by borongyuan** [comment]
We currently have no plans for ODrive Pro. Adding support for ODrive Pro won't be too difficult, but we don't want to mess up the project with more branches yet. We hope to automatically identify hardware and firmware versions in the future to reduce the number of branches, but we haven't found a better way to do so yet. Also, we need some refactoring to add CAN support and change the USB communication to be asynchronous to increase performance when using multiple hardwares (#19). Until this part of the work is done, there is little point in supporting ODrive Pro.
I think another significance of this project is to demonstrate how to develop hardware drivers in the ros2_control framework. We also demonstrated how to run ros2_control_demos on ODrive, since it happens to be a dual drive. We also used this as a template to develop ros2_control drivers for our other hardwares last year. Therefore, I suggest that you try to modify it yourself to meet different usage needs. We will also make the implementation of this project more standardized (according to the framework of ros2_control and lifecycle node).

**Comment by solomondg** [comment]
Note that there's now an official ROS2 package for ODrive Pro and S1 - https://github.com/odriverobotics/odrive_can

**Comment by RBreugelmans453** [comment]
Are there any updates on the possible support for the ODrive Pro?

**Comment by solomondg** [comment]
> Are there any updates on the possible support for the ODrive Pro?

Hi! You may want to use the official ODrive ROS2 package, which has a WIP ros2 control implementation: https://github.com/odriverobotics/ros_odrive

**Comment by RBreugelmans453** [comment]
Yes, I am aware, but at this time it is not yet possible for me to use CAN instead of USB, so I was hoping to use this package.
